src/einsteinpy/utils/schwarzschild_utils.py

```diff
@@ -21,9 +20,13 @@ def schwarzschild_radius(mass):
         Schwarzschild radius for a given mass
 
     """
+    if not isinstance(mass, u.quantity.Quantity):
+        mass = mass * u.kg
+    if not isinstance(c, u.quantity.Quantity):
+        c = c * u.m / u.s
```

Comment:
Do the same for `G`
`if not isinstance(G, u.quantity.Quantity): G = G * constant.G.unit`

After that, the change in tests wouldn't be required I guess

Reply:
still getting error

Reply:
Lemme see

Reply:
its working ..

Reply:
> still getting error

After doing the required changes, the old tests are doing fine on my system.
And another thing I noticed that tests are running a lot slower (nearly double the previous time) and I think the culprit is astropy.units aware `schwarzschild_radius()` function which I predicted earlier. 
As for the last point, leave it right now, but we may have to drop using the `schwarzschild_radius()` function altogether and define `Rs=2*M*G/(c**2)` in every function or define a private function which deals only in float.

Reply:
> > still getting error
> 
> After doing the required changes, the old tests are doing fine on my system.
> And another thing I noticed that tests are running a lot slower (nearly double the previous time) and I think the culprit is astropy.units aware `schwarzschild_radius()` function which I predicted earlier.
> As for the last point, leave it right now, but we may have to drop using the `schwarzschild_radius()` function altogether and define `Rs=2*M*G/(c**2)` in every function or define a private function which deals only in float.

@shreyasbapat @Masquerade0097 Thoughts??

Reply:
yes i too noticed it is slower now.
but when i changed the argument in max function in geodesic_static/_draw_attractor , it was working fine.

Reply:
i did this  - (schwarzschild_radius(self.mass) * 10000)*u.km , and it was working fine.

Reply:
the max function there was comparing the two quantities one of which was in astropy units and other was not.

Reply:
> the max function there was comparing the two quantities one of which was in astropy units and other was not.

It's corrected right, no longer the tests are failing

Reply:
Old tests
![24](https://user-images.githubusercontent.com/24695764/56272449-58e4a680-6118-11e9-8cda-c5d153b57fe8.PNG)
New tests
![25](https://user-images.githubusercontent.com/24695764/56272465-600bb480-6118-11e9-969c-25985bec2e30.PNG)

Reply:
Hi @alpeshjamgade 
I had an discussion with @shreyasbapat regarding the performance of the metric due to `schwarzschild_radius()` .  
So, could you make a function `schwarzschild_radius_dimensionless(M, c=constant.c.value, G=constant.G.value)` which deals only in float. Use this function everywhere wherever `Rs` needs to be calculated dimensionlessly ;) .

Reply:
> 
> 
> Hi @alpeshjamgade
> I had an discussion with @shreyasbapat regarding the performance of the metric due to `schwarzschild_radius()` .
> So, could you make a function `schwarzschild_radius_dimensionless(M, c=constant.c.value, G=constant.G.value)` which deals only in float. Use this function everywhere wherever `Rs` needs to be calculated dimensionlessly ;) .

sure!

Reply:
Make this function in `schwarzschild_utils.py`